### ISSUE TITLE
Table mini bar chart relative to a second column

### ISSUE THREAD

**Issue by MarkRx** [metabase/metabase#9206]
Right now the embedded mini bar chart in a table can only display a value relative to the other values in the same column. As a user I would like to be able to display this value relative to a value in another column in the same row.

Rules:
- Both columns must be numbers
- If the second column is < the first then an empty bar is shown
- If the first column is > the first then a full bar is shown
- If both columns are 0 then an empty bar is shown

Example: # of something complete toward some total
![image](https://user-images.githubusercontent.com/17504314/50641393-c4be4d00-0f35-11e9-8688-7c5b1344eb9b.png)

**Comment by omerfbaspinar** [comment]
thats a quite vital ! i hope they add them.

**Comment by themelinaKz** [comment]
Has an alternative been found resolving this issue? For instance if I have a percent (ex. 80) in the # Complete column is there a way to add a goal(ex. 100) to the mini bar so that the progress is relative to this goal?

**Comment by liqinatpulumi** [comment]
@themelinaKz Yes, I've found an alternative. You can just add a row and then put 100 in the field that you want to show minibar, then all other rows will show progress towards this goal. 100 has to be maximum though because the entire column is comparing against largest value of this column.

**Comment by themelinaKz** [comment]
@liqinatpulumi I was actually hoping to avoid adding any unneeded data to the database, but in any case thanks for your response

**Comment by liqinatpulumi** [comment]
You are not adding data to database, instead it's only added in query result.

**Comment by themelinaKz** [comment]
@liqinatpulumi okay I now see what you meant. However, in either case it would be preferable to not alter the final result the client sees, regardless of where the alteration has been made. Maybe when we have hundreds of rows a single one may not make much difference, but in my case I have a TOP k table in which I want to use this feature so this will not do.

**Comment by ojjj13** [comment]
I would like to bring this up as it should be a reasonable feature. It's intuitive that when people use minibar visualiztion combine with percentage notation to show a relative progress overall for that entry, it essentially act as a progress bar in a table next to the percentage, current version will fill up the minibar of the maximum value of that column, even it is not 100%. I suggest to offer an option where user can pick whether they want the minibar to act as a progress bar, thanks.

**Comment by cassie-hudson** [comment]
Another note here re: use cases where a goal is 100%, but the goal has been exceeded in some categories:

<img width="149" height="296" alt="Image" src="https://github.com/user-attachments/assets/4b19f38d-157c-46d9-bedf-96278b51d2af" />

Because the mini-bar is relative to the maximum number when over 1, these bars make it appear as if 100% is not meeting the goal when one or more values exceeds 100%.

**Comment by Bagua91** [comment]
That would be a huge if the mini-bar is not related to the maximum number.
I want to use it as a progess bar for our manufacturing.

And it would be a great if you could adjust a few things like:

- Which column should be the goal
- Or a hard written number

Thanks for your work, I love metabase <3
